src/odbc_connection.cpp

```diff
@@ -2782,108 +2840,334 @@ fetch_and_store
+                      {
+                        free(row[column_index].wchar_data);
+                        *alloc_error = true;
+                        return return_code;
+                      } 
+                      row[column_index].wchar_data = temp_realloc;
+                      row[column_index].size += data_returned_length;
+                      target_buffer =
+                        row[column_index].wchar_data + row[column_index].size;
+                    }
+                    case SQL_C_CHAR:
+                    default:
+                    {
+                      data_returned_length =
+                      strlen
+                      (
+                        (const char *) target_buffer
+                      );
+                      buffer_size =
+                        string_length_or_indicator == SQL_NO_TOTAL ?
+                        buffer_size * 2 :
+                        row[column_index].size + string_length_or_indicator + 1;
+                      SQLCHAR *temp_realloc =
+                        (SQLCHAR *)
+                        realloc
+                        (
+                          row[column_index].char_data,
+                          buffer_size
+                        );
+                      if (temp_realloc == NULL)
+                      {
+                        free(row[column_index].char_data);
+                        *alloc_error = true;
+                        return return_code;
+                      }
+                      row[column_index].char_data = temp_realloc;
+                      row[column_index].size += data_returned_length;
+                      target_buffer =
+                        row[column_index].char_data + row[column_index].size;
+                    }
+                  }
                 }
 
-              // TODO: Unhandled C types:
-              // SQL_C_SSHORT
-              // SQL_C_SHORT
-              // SQL_C_STINYINT
-              // SQL_C_TINYINT
-              // SQL_C_ULONG
-              // SQL_C_LONG
-              // SQL_C_FLOAT
-              // SQL_C_BIT
-              // SQL_C_STINYINT
-              // SQL_C_TINYINT
-              // SQL_C_SBIGINT
-              // SQL_C_BOOKMARK
-              // SQL_C_VARBOOKMARK
-              // All C interval data types
-              // SQL_C_TYPE_DATE
-              // SQL_C_TYPE_TIME
-              // SQL_C_TYPE_TIMESTAMP
-              // SQL_C_TYPE_NUMERIC
-              // SQL_C_GUID
+                return_code =
+                SQLGetData
+                (
+                  data->hstmt,
+                  column_index + 1,
+                  data->columns[column_index]->bind_type,
+                  target_buffer,
+                  buffer_size - row[column_index].size,
+                  &string_length_or_indicator
+                );
+                if (!SQL_SUCCEEDED(return_code))
+                {
+                  return return_code;
+                } 
```

Comment:
This doesn't look right to me. Shouldn't this be in side the loop above?

Reply:
Good catch, I was dealing with some bracket-hell after refactoring all of this and looks like it got put in the wrong level. Oddly, it worked ok with resizing a buffer. Will test again before push.